I found an edge case apparently in one of the tests:

```

[gw0] linux -- Python 3.11.15 /home/runner/miniconda3/envs/test-environment/bin/python3.11
    def test_time_seconds():
        """Test conversion of timecode to datetime64."""
        current = datetime.now()
>       current = current.replace(microsecond=round(current.microsecond, -3))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       ValueError: microsecond must be in 0..999999
current    = datetime.datetime(2026, 5, 20, 14, 2, 28, 999833)
satpy/tests/reader_tests/test_avhrr_l0_hrpt.py:193: ValueError
```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -72,7 +72,7 @@ jobs:
         #   compatibility issues with numpy 2. When conda-forge has numpy 2 available
         #   this shouldn't be needed.
         run: |
-          python -m pip install versioneer extension-helpers setuptools-scm configobj pkgconfig hatchling hatch-vcs
+          python -m pip install versioneer extension-helpers setuptools-scm configobj pkgconfig hatchling hatch-vcs hatch-cython
           python -m pip install \
           --index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple/ \
           --trusted-host pypi.anaconda.org \

--- a/satpy/tests/reader_tests/test_avhrr_l0_hrpt.py
+++ b/satpy/tests/reader_tests/test_avhrr_l0_hrpt.py
@@ -189,7 +189,6 @@ def to_timecode(dt_time):
 
 def test_time_seconds():
     """Test conversion of timecode to datetime64."""
-    current = datetime.now()
-    current = current.replace(microsecond=round(current.microsecond, -3))
-    timecode, year = to_timecode(current)
-    assert time_seconds(np.array([timecode], "u2"), year) == np.datetime64(current)
+    dtime = datetime(2026, 5, 20, 14, 2, 28, 999000)
+    timecode, year = to_timecode(dtime)
+    assert time_seconds(np.array([timecode], "u2"), year) == np.datetime64(dtime)

